File: 프로그래머스/Level_0/java/가까운 수.java
package 프로그래머스.Level_0.java;

import java.util.*;

class Solution {
    public int solution(int[] array, int n) {
        int answer = array[0];
        
        Arrays.sort(array);
        
        int abs = Math.abs(n-array[0]);
        int num = abs;
        
        for(int i=1; i<array.length; i++){
            int a = array[i];
            
            abs = Math.abs(n-a);
            
            if(abs < num){
                num = abs;
                answer = a;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/Level_0/가까운 수.py
def solution(array, n):
    answer = array[0]
    
    array.sort()
    
    aabs = abs(n-array[0])
    num = aabs
    
    for i in range(len(array)):
        a = array[i]
        
        aabs = abs(n-a)
        
        if(aabs < num):
            num = aabs
            answer = a
    
    return answer
# 출처: 프로그래머스 코딩 테스트 연습, https://school.programmers.co.kr/learn/challenges